src/jobs/models/manager.ts

```diff
@@ -53,11 +53,11 @@ export class JobManager {
       if (stagesReq !== undefined && stagesReq.length > 0) {
         const stages: StageCreateModel[] = stagesReq;
         stagesInput = stages.map((stage) => {
-          const { type, ...rest } = stage;
+          const { type, isWaiting, ...rest } = stage;
           const stageFull = Object.assign(rest, {
-            xstate: persistenceSnapshot,
+            xstate: getInitialXstate(stage),
             name: type,
-            status: StageOperationStatus.CREATED,
+            status: (isWaiting ?? false) ? StageOperationStatus.WAITING : StageOperationStatus.CREATED,
```

Comment:
1. I feel like you have some duplicated logic between handling stages inside jobs, to handling it on stage alone.
2. Im not sure, but didnt we talk about not allowing creating a job with stages for now, and forcing the user to create them separatly?

Reply:
it will be on next PR's after I will validating again with Nati that it ok